File: 剑指offer/每日一题/leetcode2379.cpp
#include <string>
using namespace std;
class Solution {
public:
    int minimumRecolors(string blocks, int k) {
        int l = 0;
        int r =  0;
        int count = 0;
        while(r < k){
            if(blocks[r] == 'W'){
                count += 1;
            }
            r++;
        }
        int ans = count;
        while( r < blocks.length())
        {
            if(blocks[r] == 'W'){
                count += 1;
            }
            if(blocks[l] == 'W'){
                count -= 1;
            }
            ans = min(count, ans);
            r ++;
            l ++;

        }
        return ans;
    }
};

File: 剑指offer/每日一题/leetcode2379.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        n  = len(blocks)
        ans = n
        for i in range(0, n - k + 1):
            count = 0
            for j in range(i, i + k):
                if blocks[i] == 'W':
                    count += 1
        ans = min(n, count)
        return ans
    